src/components/KSelect/KSelect.vue

```diff
@@ -713,9 +715,12 @@ watch(() => props.items, (newValue, oldValue) => {
   }
 }, { deep: true, immediate: true })
 
-watch(filterQuery, (q: string) => {
-  emit('query-change', q)
-})
+watch(filterQuery, (query: string) => {
+  if (!skipQueryChangeEmit.value || !query) {
+    emit('query-change', query)
+  }
```

Comment:
question: why would we not emit when the query is emptied? 

If a user searches, emit. If they then clear the search, why wouldn't we emit that it changed again?

Reply:
We do emit when it's empty

Reply:
Doesn't the if statement above say `|| !query` ? 

if skip is `true` it will emit, or if there's a query it will emit.

Reply:
No, if `!skip` or `!query` -> emit..

Reply:
Here is verifiation

https://github.com/user-attachments/assets/59229d33-f139-4b7e-b8e1-e607e98e67a2